metrics/handler_wrapper.go

```diff
@@ -0,0 +1,31 @@
+package metrics
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/felixge/httpsnoop"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type MetricsHandler struct {
+	Route   string
+	Handler http.Handler
+}
+
+func WrapHandler(handler http.Handler) http.Handler {
+	return MetricsHandler{
+		Handler: handler,
+	}
+}
+
+func (handler MetricsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	metrics := httpsnoop.CaptureMetrics(handler.Handler, w, r)
+
+	HttpResponseDuration.
+		With(prometheus.Labels{
+			"code":  strconv.Itoa(metrics.Code),
```

Comment:
I don't think there's a lot of value of having the plain `code` here - we could, instead, move towards `2xx`, reducing the amount of possible values that this can take down to 4 (`2xx`, `3xx`, `4xx`, `5xx`) - just what we care about